app/org/sagebionetworks/bridge/models/schedules/BaseTaskScheduler.java

```diff
@@ -0,0 +1,82 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import static org.sagebionetworks.bridge.models.schedules.ScheduleType.ONCE;
+
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalTime;
+import org.sagebionetworks.bridge.BridgeUtils;
+
+abstract class BaseTaskScheduler implements TaskScheduler {
+
+    protected final String schedulePlanGuid;
+    protected final Schedule schedule;
+    
+    BaseTaskScheduler(String schedulePlanGuid, Schedule schedule) {
+        this.schedulePlanGuid = schedulePlanGuid;
+        this.schedule = schedule;
+    }
+    
+    protected DateTime getStartTimeBasedOnEvent(Schedule schedule, Map<String, DateTime> events) {
+        // If no event is specified, it's enrollment by default.
+        String eventId = schedule.getEventId();
```

Comment:
It's not clear that eventId actually means starting event ID. Can we rename this to make it more clear?